packages/muelu/research/regionMG/src/SetupRegionHierarchy_def.hpp

```diff
@@ -736,31 +721,30 @@ void MakeInterfaceScalingFactors(const int maxRegPerProc,
 
 
 template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-void createRegionHierarchy(const int maxRegPerProc,
-                           const int numDimensions,
-                           const Array<Array<int> > lNodesPerDim,
-                           const Array<std::string> aggregationRegionType,
+void createRegionHierarchy(const int numDimensions,
+                           const Array<int> lNodesPerDim,
+                           const std::string aggregationRegionType,
                            RCP<Teuchos::ParameterList>& interfaceParams,
                            const std::string xmlFileName,
-                           Array<RCP<Xpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node> > >& nullspace,
-                           Array<RCP<Xpetra::MultiVector<typename Teuchos::ScalarTraits<Scalar>::coordinateType, LocalOrdinal, GlobalOrdinal, Node> > >& coordinates,
-                           std::vector<RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> > >& regionGrpMats,
+                           RCP<Xpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node> >& nullspace,
+                           RCP<Xpetra::MultiVector<typename Teuchos::ScalarTraits<Scalar>::coordinateType, LocalOrdinal, GlobalOrdinal, Node> >& coordinates,
+                           RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> >& regionGrpMats,
```

Comment:
The removed `std::vector` was related to the concept of "groups". Hence, this could be renamed to 
```suggestion
                           RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> >& regionMats,
```